libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m

```diff
@@ -471,25 +471,77 @@ - (NSString *)devInfoTitleString
 
 - (NSArray<SFSDKDevAction *>*) getDevActions:(UIViewController *)presentedViewController
 {
-    return @[
-             [[SFSDKDevAction alloc]initWith:@"Show dev info" handler:^{
-                 SFSDKDevInfoViewController *devInfo = [[SFSDKDevInfoViewController alloc] init];
-                 [presentedViewController presentViewController:devInfo animated:NO completion:nil];
-             }],
-             [[SFSDKDevAction alloc]initWith:@"Logout" handler:^{
-                 [[SFUserAccountManager  sharedInstance] logout:SFLogoutReasonUserInitiated];
-             }],
-             [[SFSDKDevAction alloc]initWith:@"Switch user" handler:^{
-                 SFDefaultUserManagementViewController *umvc = [[SFDefaultUserManagementViewController alloc] initWithCompletionBlock:^(SFUserManagementAction action) {
-                     [presentedViewController dismissViewControllerAnimated:YES completion:nil];
-                 }];
-                 [presentedViewController presentViewController:umvc animated:YES completion:nil];
-             }],
-             [[SFSDKDevAction alloc]initWith:@"Inspect Key-Value Store" handler:^{
-                 UIViewController *keyValueStoreInspector = [[SFSDKKeyValueEncryptedFileStoreViewController new] createUI];
-                 [presentedViewController presentViewController:keyValueStoreInspector animated:YES completion:nil];
-             }]
-    ];
+    NSMutableArray<SFSDKDevAction *> *actions = [NSMutableArray array];
+    SFUserAccountManager *userAccountManager = [SFUserAccountManager sharedInstance];
+    SFUserAccount *currentUser = userAccountManager.currentUser;
+    
+    // Check if we're showing the login screen
+    BOOL isShowingLogin = [presentedViewController isKindOfClass:[SFLoginViewController class]];
+    // TODO uncomment to support advanced auth case (once we add code to restart auth in that case below)
+    // || [presentedViewController.presentingViewController isKindOfClass:[SFSDKAuthRootController class]];
+    
+    // Show dev info - always available
+    [actions addObject:[[SFSDKDevAction alloc]initWith:@"Show dev info" handler:^{
+        UIViewController *devInfo = [SFSDKDevInfoViewController makeViewController];
+        [presentedViewController presentViewController:devInfo animated:YES completion:nil];
+    }]];
+    
+    // Login Options - only show on login screen
+    if (isShowingLogin) {
```

Comment:
NIT: Is there a reason to _only_ show it on the login screen?

Reply:
It's only useful if you do a login right after - and if you do a login (let's say after a logout or when adding a new user), then you will get a chance to bring up the screen when the login screen is shown. So it feels like login is the right time to show the login options screen. The oauth settings should be inspectable through dev infos at any time though.